### PR TITLE
Render templates directly without Mustache

### PR DESCRIPTION
**Pull request by sharat87**

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java [hunk 1]
```diff
@@ -281,12 +275,18 @@ private static void clearAndPushToken(StringBuilder tokenBuilder, List<String> t
...
-    private static String render(String template, String name, Map<String, String> keyValueMap) {
-        MustacheFactory mf = new DefaultMustacheFactory();
-        Mustache mustache = mf.compile(new StringReader(template), name);
-        Writer writer = new StringWriter();
-        mustache.execute(writer, keyValueMap);
-        return StringEscapeUtils.unescapeHtml4(writer.toString());
+    public static String render(String template, Map<String, String> keyValueMap) {
+        final StringBuilder rendered = new StringBuilder();
+
+        for (String token : tokenize(template)) {
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:281]: Possibly not related but I wanted to ask, what happens if the binding is returning a JSON object. Eg:
```
{{
Api.data.map((row) => {
return {
  key: row.value
}
})
}}
```

In this scenario, the `}}` happens inside the map function and also at the end to close the binding. Will our tokenizer be able to catch this scenario?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:281]: I believe it is converted to JSON much before it gets to this point. The map's generic type says the values can only ever be strings, so there shouldn't be a change in behavior there. Whatever was passing strings to this function should keep doing it.

Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:280]: Can we add a test for this scenario as well? Just to be sure that the server does the replacements properly in this use-case as well?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:280]: I just checked and it turns out the client sends us the stringified value in advance. So, we don't accept/expect anything else here.

Screenshot 2020-12-03 at 13 46 26

Does that sound fine?

Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:280]: For my understanding, what does the client send on the wire for the scenario I mentioned? I'm worried about `}}` appearing pre-maturely in the binding itself. I just want to be doubly-sure that we have handled this scenario in our tokenizer.

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/MustacheHelper.java:280]: Tests checking for that particular case in the tokenizer are already there: https://github.com/appsmithorg/appsmith/blob/1fc7cbe3afbe8ccb3664f57b27cd2fab58086529/app/server/appsmith-server/src/test/java/com/appsmith/server/helpers/MustacheHelperTest.java#L155

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**